packages/perseus/src/widgets/categorizer/categorizer-validator.test.ts

```diff
@@ -0,0 +1,185 @@
+
+describe("categorizer widget", () => {
+    let userEvent: UserEvent;
+    beforeEach(() => {
+        userEvent = userEventLib.setup({
+            advanceTimers: jest.advanceTimersByTime,
+        });
+
+        jest.spyOn(Dependencies, "getDependencies").mockReturnValue(
+            testDependencies,
+        );
+    });
+
+    it("is incorrect when blank", async () => {
+        // Arrange
+        const apiOptions: APIOptions = {
+            isMobile: false,
+        };
+        const {renderer} = renderQuestion(question1, apiOptions);
+
+        // Act
+        const [_, score] = renderer.guessAndScore();
+
+        // Assert
+        expect(score).toMatchInlineSnapshot(`
+            {
+              "message": "Make sure you select something for every row.",
+              "type": "invalid",
+            }
+        `);
+    });
+
+    it("can be answered incorrectly", async () => {
+        // arrange
+        const {renderer} = renderQuestion(question1);
+
+        const firstItem = screen.getAllByRole("row")[0];
+        await userEvent.click(firstItem);
+
+        // act
+        const [_, score] = renderer.guessAndScore();
+
+        // assert
+        expect(score).toMatchInlineSnapshot(`
+            {
+              "message": "Make sure you select something for every row.",
+              "type": "invalid",
+            }
+        `);
+    });
+
+    it("can be answered correctly", async () => {
+        // arrange
+        const {renderer} = renderQuestion(question1);
+
+        // act
+        await userEvent.click(
+            screen.getAllByRole("button", {name: "No relationship"})[0],
+        );
+        await userEvent.click(
+            screen.getAllByRole("button", {
+                name: "Positive linear relationship",
+            })[0],
+        );
+        await userEvent.click(
+            screen.getAllByRole("button", {
+                name: "Negative linear relationship",
+            })[1],
+        );
+        await userEvent.click(
+            screen.getAllByRole("button", {
+                name: "Nonlinear relationship",
+            })[1],
+        );
+
+        renderer.guessAndScore();
+
+        // assert
+        expect(renderer).toHaveBeenAnsweredCorrectly();
+    });
+});
+
+describe("validating answers", () => {
+    it("gives points when the answer is correct", () => {
+        const rubric: Rubric = {
+            items: ["Graph $1$", "Graph $2$"],
+            values: [1, 3],
+            randomizeItems: false,
+            categories: [
+                "No relationship",
+                "Positive linear relationship",
+                "Negative linear relationship",
+                "Nonlinear relationship",
+            ],
+            highlightLint: false,
+            static: false,
+        };
+
+        const useInput = {
+            values: [1, 3],
+        } as const;
+        const score = Categorizer.validate(useInput, rubric, mockStrings);
```

Comment:
We need to replace all instances of `[Widget].validate` and `[Widget].simpleValidate` with the pure `widgetValidator` function. No need to go looking for uses, but in files we touch we should be making this change.